docs/DeveloperGuide.adoc

```diff
@@ -153,7 +153,175 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+==== Delete command
+
+The following is a detailed explanation of the operations `QuestionDeleteCommand` performs. +
+
+*Step 1*. The `QuestionDeleteCommand#execute(Model model)` method is executed and it checks if the `Index` defined when
+instantiating `QuestionDeleteCommand(Index index, HashMap<String, String> fields)` is valid.
+
+*Step 2*. The `Question` at the specified `Index` is then removed from the `QuestionBank#questions` observable list through
+the `Model#deleteQuestion(Index index)` method.
+
+*Step 3*. A success message will be generated by the `QuestionAddCommand#generateSuccessMessage(Question question)` method
+and a new `CommandResult` will be returned with the generated success message.
+
+[[Feature-Question-List]]
+==== List command
+
+The following is a detailed explanation of the operations `QuestionListCommand` performs. +
+
+*Step 1*. The `QuestionListCommand#execute(Model model)` method is executed. No validation is necessary here since it
+does not write to the question list.
+
+*Step 2*. The `Model#getQuestionsSummary()` method is then called and the questions are returned as a `String` to the new `CommandResult`
+object.
+
+==== Design Considerations
+
+===== Command Syntax
+* ** Current Implementation: **
+** Current implementation of the commands follows the command word syntax e.g question followed by the arguments necessary
+to execute the add, edit, delete, list and slideshow command.
+
+* ** Alternatives Considered: **
+** Usage of a forward slash `/` and then the command word. Although it makes it clearer that the input is a command, we
+realised it is redundant as the only input will be commands and will make it more tedious for the user.
+
+===== Command Length
+
+* ** Current Implementation: **
+** Commands are currently shortened as much as possible through the use of initials without much loss in clarity.
+For example, instead of using optionA/ to denote the first option, we use a/ instead. Although this may be unfamiliar
+to the user initially, it should be easy to pick up and will make it less tedious during input.
+
+* ** Alternatives Considered: **
+** Using more descriptive terms as arguments such that each argument will be specified clearly. However, this decreases
+the user experience as the command will be too long.
+
+[[Feature-Slideshow]]
+=== Slideshow feature
+
+The slideshow feature is dependent on the questions added by the `question slideshow [question no(s).]` command and
+interacts with the `ModelManager` to retrieve the list of questions to be displayed in the slideshow.
+The logic control for displaying the ui resides in the `SlideshowWindow` class and handles the controls and instantiation
+of the various `QuestionPanel` that contains each question. +
+
+Below is the sequence diagram of the interactions that happen from when the slideshow command is entered, to the corresponding
+questions displayed in the slideshow.
+
+image::SlideshowFeatureSequenceDiagram.png[]
```

Comment:
Nice diagram! It could use a bit more exposition on what happens to the new `QuestionPanel`s. Are they added to the UI scene graph anywhere?

Reply:
Also, this diagram is quite large. Would it be worth breaking it up into two diagrams, by using `ref` and `sd` blocks?

Reply:
Is Logic part of the UI? Would be clearer if it uses different colour.

Reply:
Might want to include a caption for the diagram